packages/realm-server/tests/permissions/permission-checker-test.ts

```diff
@@ -1,41 +1,118 @@
+import { MatrixClient } from '@cardstack/runtime-common/matrix-client';
 import RealmPermissionChecker from '@cardstack/runtime-common/realm-permission-checker';
 import { module, test } from 'qunit';
 
+let matrixUserProfile: { displayname: string } | undefined = undefined;
+let mockMatrixClient = {
+  async getProfile(_userId) {
+    return matrixUserProfile;
+  },
+} as MatrixClient;
 module('realm-user-permissions', function (_hooks) {
   module('world-readable realm', function () {
-    let permissionsChecker = new RealmPermissionChecker({
-      '*': ['read'],
-    });
+    let permissionsChecker = new RealmPermissionChecker(
+      {
+        '*': ['read'],
+      },
+      mockMatrixClient,
+    );
+
+    test('anyone can read but not write', async function (assert) {
+      assert.ok(await permissionsChecker.can('anyone', 'read'));
+      assert.notOk(await permissionsChecker.can('anyone', 'write'));
 
-    test('anyone can read but not write', function (assert) {
-      assert.ok(permissionsChecker.can('anyone', 'read'));
-      assert.notOk(permissionsChecker.can('anyone', 'write'));
+      assert.deepEqual(await permissionsChecker.for('anyone'), ['read']);
     });
   });
 
   module('world-writable realm', function () {
-    let permissionsChecker = new RealmPermissionChecker({
-      '*': ['read', 'write'],
+    let permissionsChecker = new RealmPermissionChecker(
+      {
+        '*': ['read', 'write'],
+      },
+      mockMatrixClient,
+    );
+
+    test('anyone can read and write', async function (assert) {
+      assert.ok(await permissionsChecker.can('anyone', 'read'));
+      assert.ok(await permissionsChecker.can('anyone', 'write'));
+
+      assert.deepEqual(await permissionsChecker.for('anyone'), [
+        'read',
+        'write',
+      ]);
     });
+  });
+
+  module('users-readable realm', function () {
+    let permissionsChecker = new RealmPermissionChecker(
+      {
+        users: ['read', 'write'],
+        '@matic:boxel-ai': ['read', 'write'],
```

Comment:
let's change this test so that `users` only has `['read']` permissions. This way we can confirm that `@matic:boxel-ai`'s permissions are the union of the `users` permissions and `@matic:boxel-ai` permissions (`['read', 'write']`), and that `@not-matic:boxel-ai` (which i assume is a matrix user) has only `['read']`, and that a non matrix user (aka anonymous access) as no permissions.